react/week4/git_user_search/src/GitUserSearch.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { useDebounce } from 'use-debounce';
+import './App.css';
+import img from './images/github.png';
+
+const GitUserSearch = () => {
+  const [searchValue, setSearchValue] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+  const [debouncedSearchValue] = useDebounce(searchValue, 1000);
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const myRef = useRef();
+
+  const onChange = (e) => {
+    setSearchValue(e.target.value);
+    setLoading(true);
+  };
+
+  useEffect(() => {
+    if (debouncedSearchValue !== '') {
+      fetch(`https://api.github.com/search/users?q=${debouncedSearchValue}`) //debouncing-to deal with api time-outs
+        .then((results) => {
+          if (results.ok) {
+            setError(false);
+            return results.json();
+          } else {
+            setError(true);
```

Comment:
Error handing could be better. Code breaks right now if i input some random characters. Try to use "try" and "Catch" for better error handling and it will help in below points as well

* Right now it shows both messages if i search something
* And when i empty the searchbox it still shows loading


![Screenshot_2020-11-19 React App](https://user-images.githubusercontent.com/55982058/99684148-6b0b9100-2a81-11eb-9cdc-ea2680b94c9f.png)